docs/sources/userguide/plugins.md

```diff
@@ -0,0 +1,157 @@
+Docker Plugins MVP
+==================
+
+Plugins provide a mechanism for hooking into the Docker engine to extend its behavior.
+
+HTTP over a local UNIX socket is used to communicate between Docker and its plugins.
+
+Plugins are distributed as containers.
```

Comment:
+1 for having **some** way for a plugin to speak to the main docker server - it is vital for various reasons - I'm not overly concerned with how though (`-v` or `--allow-docker-api`)